### PR TITLE
[VPDQ] Make ffmpeg path optional

### PR DESCRIPTION
**Pull request by tigerchensh**
Summary
---------

Invofe ffmpeg by using $PATH 'ffmpeg' in CPP implementation instead of binary's address. So, the ffmpeg's path is optional if it is installed properly.

Test Plan
---------

Run through the regression test without ffmpeg path argument.
python3 regtest.py -d ../ThreatExchange/vpdq/output-hashes -i ../ThreatExchange/tmk/sample-videos

### CODE REVIEW

File: vpdq/cpp/bin/vpdq-hash-video.cpp [hunk 1]
```diff
@@ -9,11 +9,11 @@ static void usage(char* argv0, int rc) {
   FILE* fp = (rc == 0) ? stdout : stderr;
   fprintf(fp, "Usage: %s [options]\n", argv0);
   fprintf(fp, "Required:\n");
-  fprintf(fp, "-f|--ffmpeg-path ...\n");
   fprintf(fp, "-i|--input-video-file-name ...\n");
   fprintf(fp, "-o|--output-hash-file-name ...\n");
   fprintf(fp, "-r|--seconds-per-hash ...:Must be a positive integer\n");
   fprintf(fp, "Options:\n");
+  fprintf(fp, "-f|--ffmpeg-path: Specific ffmpeg's address you want to use\n");
```
Comment (BarrettOlson) [inline vpdq/cpp/bin/vpdq-hash-video.cpp:16]: nit: wording "path to ffmpeg" instead of "ffmpeg's address"

**Review by BarrettOlson — CHANGES_REQUESTED** [review]
Overall looks good. The one blocking concern I have is around error handling.

Can you add a call to `regtest.py` to the CI?

Also I think it is worth checking the error message when `ffmpeg` cannot be found on the path. (you should be able to check this by hiding/overriding the path before running the test locally).

If we error isn't clear/helpful we should add a check for catching this at the start and print out the issue clearly.

**Comment by tigerchensh** [comment]
The error message will be "sh: ffmpeg: command not found" for no ffmpeg installed.

File: vpdq/cpp/regtest.py [hunk 2]
```diff
@@ -8,7 +8,13 @@ def get_argparse() -> argparse.ArgumentParser:
     ap = argparse.ArgumentParser(
         description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
     )
-    ap.add_argument("-f", "--ffmpegPath", metavar="FFMPEG_PATH", help="ffmpeg path")
+    ap.add_argument(
+        "-f",
+        "--ffmpegPath",
+        metavar="FFMPEG_PATH",
```
Comment (Dcallies) [inline vpdq/cpp/regtest.py:14]: (nit, ignorable): this is very close to the default, might be fine to ignore it.

File: vpdq/cpp/regtest.py [hunk 3]
```diff
@@ -8,7 +8,13 @@ def get_argparse() -> argparse.ArgumentParser:
     ap = argparse.ArgumentParser(
         description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
```
Comment (Dcallies) [inline vpdq/cpp/regtest.py:9]: Any reason to use RawDescriptionHelpFormatter? This is usually if you are doing custom things with indenting

Comment (tigerchensh) [reply vpdq/cpp/regtest.py:9]: Research the formatter_class a little bit, decide to remove it.

**Pull request merged**